### DISCUSSION TITLE
GPU Acceleration?

### DISCUSSION THREAD

**Discussion by Wamy-Dev** [EmulatorJS/EmulatorJS#458] [Ideas]
Is GPU Acceleration possible?

Found this library that might make it easier: https://github.com/gpujs/gpu.js
I can try checking it out myself.

**Comment by ethanaobrien** [comment]
Not in the way you provided. The emulator itself is written in C/C++ and this gpu.js project seems to optimize javascript code. The way we render now is probably as optimized as you can get with something like this

**Reply by Wamy-Dev** [reply]
Ah, I get it. Thank you for the insight. Even on my system, Super Mario 64, runs at about 10fps which is not great.

**Reply by ethanaobrien** [reply]
That sounds like a problem with your computer, on mine it runs at 60fps flawlessly only using about 20% of the cpu, and it's a chromebook. Would you mind sharing what type of device you have?

**Comment by Wamy-Dev** [comment]
Im on linux, Ryzen 5 3600, 32gb Ram

**Reply by ethanaobrien** [reply]
This is strange. What browser are you using? Do you have a dedicated gpu?

**Reply by Wamy-Dev** [reply]
Using Brave, I have an RTX 3060

**Reply by ethanaobrien** [reply]
This is very strange. Would you mind running it while having the system monitor open (or whatever it's called on Linux) and let me know if anything is maxed out? Could you possibly try another browser?

**Reply by Wamy-Dev** [reply]
tested a few days later, seems to not be affected anymore, I don't know why that was happening.
